x-pack/plugins/enterprise_search/public/applications/__mocks__/kea.mock.ts

```diff
@@ -58,3 +62,62 @@ export const setMockValues = (values: object) => {
 export const setMockActions = (actions: object) => {
   (useActions as jest.Mock).mockImplementation(() => ({ ...mockAllActions, ...actions }));
 };
+
+/**
+ * Kea logic helpers
+ *
+ * Call this function to mount a logic file and optionally override default values.
+ * Automatically DRYs out a lot of cruft for us, such as resetting context, creating the
+ * nested defaults path obj (see https://kea.js.org/docs/api/context#resetcontext), and
+ * returning an unmount function
+ *
+ * Example usage:
+ *
+ * import { LogicMounter } from '../../../__mocks__/kea.mock';
+ * import { SomeLogic } from './';
+ *
+ * const { mount, unmount } = new LogicMounter(SomeLogic);
+ *
+ * it('some test', () => {
+ *   mount({ someValue: 'hello' });
+ *   unmount();
+ * });
+ */
+import { resetContext, Logic, LogicInput } from 'kea';
+
+interface LogicFile {
+  inputs: Array<LogicInput<Logic>>;
+  mount(): Function;
+}
+export class LogicMounter {
+  private logicFile: LogicFile;
+  private unmountFn!: Function;
+
+  constructor(logicFile: LogicFile) {
+    this.logicFile = logicFile;
+  }
+
+  // Reset context with optional default value overrides
+  public resetContext = (values?: object) => {
+    if (!values) {
+      resetContext({});
+    } else {
+      const path = this.logicFile.inputs[0].path as string[];
+      const defaults = path.reduceRight((value: object, key: string) => ({ [key]: value }), values);
```

Comment:
Example path:
```
['enterprise_search', 'app_search', 'engine_overview_logic']
```

Example defaults:
```
{
  defaults: {
    enterprise_search: {
      app_search: {
        engine_overview_logic: {
           ...values
        }
      }
    }
  }
}
```

I've never actually seen/used `.reduceRight` before this but was super surprised to find it was that easy to write a one-liner to do exactly what we wanted 🤯

Reply:
I love me some reduce 👍